cmd/restic/cmd_check.go

```diff
@@ -349,6 +394,23 @@ func selectRandomPacksByPercentage(allPacks map[restic.ID]int64, percentage floa
 		id := keys[idx[i]]
 		packs[id] = allPacks[id]
 	}
+	return packs
+}
 
+func selectRandomPacksByFileSize(allPacks map[restic.ID]int64, subsetSize int64, repoSize int64) map[restic.ID]int64 {
+	subsetPercentage := (float64(subsetSize) / float64(repoSize)) * 100.0
+	packs := selectRandomPacksByPercentage(allPacks, subsetPercentage)
 	return packs
 }
+
+func determineReadDataSubsetType(readDataSubset string) (DataSubsetType, error) {
```

Comment:
I think this method should be inlined. Parsing everything twice only looks like it adds more code.